### PR TITLE
Droth 2300 tierekisteri data import batches failing

### PR DESCRIPTION
**Pull request by ricardofmsimoes**
Fix the following batchs:
  - TiereskiteriDataImporter
  - ValluXML
  - AssetValidatorProcess

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/ManoeuvreService.scala [hunk 1]
```diff
@@ -335,40 +342,47 @@ class ManoeuvreService(roadLinkService: RoadLinkService, eventBus: DigiroadEvent
...
+      else
+        Seq(createWithoutTransaction("traffic_sign_generated", NewManoeuvre(Set(), Seq.empty[Int], None, roadLinks.map(_.linkId), Some(trafficSignInfo.id), false), roadLinks))
 
-    if(!validateManoeuvre(trafficSignInfo.roadLink.linkId, roadLinks.last.linkId, ElementTypes.FirstElement))
-      throw new ManoeuvreCreationException(Set("Manoeuvre creation not valid"))
-    else
-      Seq(createWithoutTransaction("traffic_sign_generated", NewManoeuvre(Set(), Seq.empty[Int], None, roadLinks.map(_.linkId), Some(trafficSignInfo.id), false), roadLinks))
+    } catch {
+      case mce: ManoeuvreCreationException =>
+          insertInaccurateAsset(trafficSignInfo, fromTrafficSignGenerator, mce)
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/ManoeuvreService.scala]: What is the reason to get this exception? Something previous seems to allow incorrect operation

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/ManoeuvreService.scala [hunk 2]
```diff
@@ -335,40 +342,47 @@ class ManoeuvreService(roadLinkService: RoadLinkService, eventBus: DigiroadEvent
...
 
+      case eipe: InvalidParameterException =>
+            insertInaccurateAsset(trafficSignInfo, fromTrafficSignGenerator, eipe)
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/ManoeuvreService.scala]: What is the reason to get this exception? Something previous seems to allow incorrect operation

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/ManoeuvreService.scala [hunk 3]
```diff
@@ -322,7 +321,15 @@ class ManoeuvreService(roadLinkService: RoadLinkService, eventBus: DigiroadEvent
     }
   }
 
-  private def createManoeuvreFromTrafficSign(trafficSignInfo: TrafficSignInfo): Seq[Long] = {
+  private def insertInaccurateAsset(trafficSignInfo: TrafficSignInfo, fromTrafficSignGenerator: Boolean, exception: Exception ): Seq[Long] = {
+    if (!fromTrafficSignGenerator )
+      throw exception
+
+    inaccurateDAO.createInaccurateAsset(trafficSignInfo.id, Manoeuvres.typeId, trafficSignInfo.roadLink.municipalityCode, trafficSignInfo.roadLink.administrativeClass )
```
Comment (marquesfm) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/ManoeuvreService.scala]: The asset Id to insert at inaccurate working list need to be the same type as the assetTypeId

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/ManoeuvreService.scala [hunk 4]
```diff
@@ -322,7 +321,15 @@ class ManoeuvreService(roadLinkService: RoadLinkService, eventBus: DigiroadEvent
     }
   }
 
-  private def createManoeuvreFromTrafficSign(trafficSignInfo: TrafficSignInfo): Seq[Long] = {
+  private def insertInaccurateAsset(trafficSignInfo: TrafficSignInfo, fromTrafficSignGenerator: Boolean, exception: Exception ): Seq[Long] = {
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/ManoeuvreService.scala]: fromTrafficSignGenerator could be rename to fromTierekisteri

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/client/tierekisteri/importer/TrafficSignTierekisteriImporter.scala [hunk 5]
```diff
@@ -223,7 +221,13 @@ class TrafficSignTierekisteriImporter extends TierekisteriAssetImporterOperation
               } else
                 allowedAdditionalPanels
 
-              createPointAsset(ra, roadlink.get, mValue, trAssetData, allowedProperties.toSet)
+              //Remove additonal Panels with 'trafficSigns_type' with empty value
+              val additionalPanelsFiltered = allowedAdditionalPanels.filterNot{ panel =>
+                                            val prop = trafficSignService.getProperty(panel.propertyData, trafficSignService.typePublicId).get
```
[the remark this answers is not in the document]
Comment (ricardofmsimoes) [reply digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/client/tierekisteri/importer/TrafficSignTierekisteriImporter.scala]: Done

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/client/tierekisteri/importer/TrafficSignTierekisteriImporter.scala [hunk 6]
```diff
@@ -223,7 +221,13 @@ class TrafficSignTierekisteriImporter extends TierekisteriAssetImporterOperation
...
+                                            prop.propertyValue.toString.trim.isEmpty }
+
+              if ( trafficSignType.toString.trim.nonEmpty )
+                createPointAsset(ra, roadlink.get, mValue, trAssetData, additionalPanelsFiltered)
```
[the remark this answers is not in the document]
Comment (ricardofmsimoes) [reply digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/client/tierekisteri/importer/TrafficSignTierekisteriImporter.scala]: Fixed

File: ci/pipeline/run_manual_valluXML_conversion [hunk 7]
```diff
@@ -0,0 +1,88 @@
+import java.net.URI
```
[the remark this answers is not in the document]
Comment (ricardofmsimoes) [reply ci/pipeline/run_manual_valluXML_conversion]: Done.

[the remark this answers is not in the document]
Comment (ricardofmsimoes) [reply ci/pipeline/run_manual_valluXML_conversion]: Done

File: ci/pipeline/run_manual_valluXML_test_conversion [hunk 8]
```diff
@@ -0,0 +1,88 @@
+import java.net.URI
```
[the remark this answers is not in the document]
Comment (ricardofmsimoes) [reply ci/pipeline/run_manual_valluXML_test_conversion]: Done.

[the remark this answers is not in the document]
Comment (ricardofmsimoes) [reply ci/pipeline/run_manual_valluXML_test_conversion]: Done

File: ci/pipeline/run_valluXML_conversion [hunk 9]
```diff
@@ -16,8 +16,8 @@ pipeline {
                        string(credentialsId: 'TN_ITS_CONVERSION', variable: 'TN_ITS_CONVERSION')]) {
                             script {
                                 STAGE_RUN_ROSATTE_CONVERSION = [
-                                    "mkdir -p ${WORKSPACE}/dev/fd/ &&",
-                                    "curl -s -v -X POST https://${TN_ITS_CONVERSION}:${TN_ITS_CONVERSION}@tn-its.herokuapp.com/conversion/mass_transit_stop_on_vallu | tee ${WORKSPACE}/dev/fd/3 | tail -n 1 | grep OK"
+                                    "exec 3>&1 &&",
+                                    "curl -s -v -X POST https://${TN_ITS_CONVERSION}:${TN_ITS_CONVERSION}@tn-its.herokuapp.com/conversion/mass_transit_stop_on_vallu | tee /dev/fd/3 | tail -n 1 | grep OK"
```
[the remark this answers is not in the document]
Comment (ricardofmsimoes) [reply ci/pipeline/run_valluXML_conversion]: Done.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/middleware/TrafficSignManager.scala [hunk 10]
```diff
@@ -89,22 +87,26 @@ case class TrafficSignManager(manoeuvreService: ManoeuvreService, roadLinkServic
...
-          insertTrafficSignToProcess(trSign.id, ParkingProhibition, Some(trSign))
+          insertTrafficSignToProcess(trSign.id, ParkingProhibition, Some(trSign), newTransaction)
 
         case _ => None
       }
     }
   }
 
-  def insertTrafficSignToProcess(id: Long, assetInfo: AssetTypeInfo, persistedTrafficSign: Option[PersistedTrafficSign] = None) : Unit = {
+  def insertTrafficSignToProcess(id: Long, assetInfo: AssetTypeInfo, persistedTrafficSign: Option[PersistedTrafficSign] = None, newTransaction: Boolean = false) : Unit = {
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/middleware/TrafficSignManager.scala]: this change could be reverted?

Comment (ricardofmsimoes) [reply digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/middleware/TrafficSignManager.scala]: Due the possibility of nasted transaction, we should keep this change.
If this function is called after TrafficSignTierekisteriImporter.expireAssets we will create a nasted transaction and with this change we can avoid the issue

**Pull request merged**